### PR TITLE
Adds directory traversal for Http.Dir("/")

### LINKED ISSUE #569: Detect directory traversal http.Dir("/")
### Summary 

By following an example like https://golang.org/pkg/net/http/#NewFileTransport and allowing a user to traverse the file system from "/", it could be quite a security issue. For e.g. /proc/*/environ could be downloaded

### Steps to reproduce the behavior

https://github.com/kaihendry/badpath

### gosec version

Version: 2.6.1
Git tag: v2.6.1
Build date: 2021-01-22T10:39:07Z

### Go version (output of 'go version')

go version go1.15.8 linux/amd64

### Operating system / Environment

I use Arch btw

### Expected behavior

Maybe a warning?

### Actual behavior

No issue

### PR DESCRIPTION
**Pull request by capitanu**
Fixes #569

### CODE REVIEW

File: report/formatter_test.go [hunk 1]
```diff
@@ -276,10 +276,10 @@ var _ = Describe("Formatter", func() {
 	})
 	Context("When using different report formats", func() {
 		grules := []string{
-			"G101", "G102", "G103", "G104", "G106",
-			"G107", "G109", "G110", "G201", "G202", "G203", "G204",
-			"G301", "G302", "G303", "G304", "G305", "G401", "G402",
-			"G403", "G404", "G501", "G502", "G503", "G504", "G505",
+			"G101", "G102", "G103", "G104", "G106", "G107", "G109",
+			"G110", "G110", "G201", "G202", "G203", "G204", "G301",
```
Comment (ccojocar) [inline report/formatter_test.go:280]: Shouldn't be G111 here?

File: rules/directory-traversal.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+type traversal struct {
+	gosec.MetaData
+}
+
+func (r *traversal) ID() string {
+	return r.MetaData.ID
+}
+
+func (r *traversal) Match(n ast.Node, ctx *gosec.Context) (*gosec.Issue, error) {
+	return gosec.NewIssue(ctx, n, r.ID(), r.What, r.Severity, r.Confidence), nil
```
Comment (ccojocar) [inline rules/directory-traversal.go:18]: Are you planning to implement this rule?

Comment (capitanu) [reply rules/directory-traversal.go:18]: Yes, this is currently just a draft, will get it working in the upcoming days.

File: report/formatter_test.go (hunk 1 above)
Comment (capitanu) [reply report/formatter_test.go]: Fixed now :D

**Review by ccojocar — APPROVED** [review]
Thanks for contributing this rule. LGTM, please could you fix the lint warning?

**Comment by capitanu** [comment]
> Thanks for contributing this rule. LGTM, please could you fix the lint warning?

Thank you for the review! I will push a fix in a few minutes

**Pull request merged**